pkg/cache/controller_test.go

```diff
@@ -21,29 +21,38 @@ import (
 
 	"github.com/kcp-dev/logicalcluster/v2"
 	"github.com/stretchr/testify/require"
-	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
-	"k8s.io/apimachinery/pkg/labels"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-func newUnstructured(cluster, namespace, name string, labels labels.Set) *unstructured.Unstructured {
-	u := new(unstructured.Unstructured)
-	u.SetAnnotations(map[string]string{
-		logicalcluster.AnnotationKey: cluster,
-	})
-	u.SetNamespace(namespace)
-	u.SetName(name)
-	u.SetLabels(labels)
-	return u
-}
-
 func TestClusterIndexFunc(t *testing.T) {
 	tests := map[string]struct {
-		obj     *unstructured.Unstructured
+		obj     *metav1.ObjectMeta
 		desired string
 	}{
-		"bare cluster":             {obj: newUnstructured("test", "", "name", nil), desired: "test//"},
-		"cluster and namespace":    {obj: newUnstructured("test", "namespace", "name", nil), desired: "test//"},
-		"bare cluster with dashes": {obj: newUnstructured("test-with-dashes", "", "name", nil), desired: "test-with-dashes//"},
```

Comment:
This is a personal nit but having to read helpers to figure out what the order of these arguments means, along with having to figure out that unstructured & `nil` labels were red herrings, is not ideal for comprehension of a test IMO